tests/integration/targets/test_ansible_type/tasks/tasks.yml

```
@@ -0,0 +1,248 @@
+# Copyright (c) Ansible Project
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+# Substitution converts str to AnsibleUnicode
+# -------------------------------------------
+
+- name: String. AnsibleUnicode.
+  assert:
+    that: data is community.general.ansible_type(dtype)
+    success_msg: '"abc" is {{ dtype }}'
+    fail_msg: '"abc" is {{ result }}'
+    quiet: '{{ quiet_test | d(true) | bool }}'
+  vars:
+    data: "abc"
+    result: '{{ data | community.general.reveal_ansible_type }}'
+    dtype: 'AnsibleUnicode'
```

It would be simpler and arguably more readable as:
```suggestion
- name: String. AnsibleUnicode.
  assert:
    that: >
      "abc" is community.general.ansible_type("AnsibleUnicode")
```
but that is my personal preference. Same for following tests: keep it simple.

---

well, personally I still think the test code is more verbose than it needs to be, but let's go with it as-is, and we may review it later.

---

Yes, let's review this later.

JFYI, the integration test *tasks.yml* and the *Examples* were generated from the *tests* list from the (removed) integration test vars. I've prepared the next PR to add a chapter in *docsite*. The RST file is also generated from the *tests* list by the helper *docs/docsite/helper/ansible_type*. Let's consolidate the docsite, examples, and tests later. I want to keep them consistent. The challenge is the AnsibleUnicode/str dependency on the substitution.